### ISSUE TITLE
iOS 8 Crashes on Deeplink

### ISSUE THREAD

**Issue by Bobcat64** [BranchMetrics/xamarin-branch-deep-linking#37]
Is iOS 8 supported with Deeplinks? I appear to be running into a crash if I tap on one of our deeplinks and then install and run the app for the first time. For debugging I uninstalled the app, tapped the link, and then ran the project.

The exception is as follows:
`NSInvalidArgumentException Reason: *** setObjectForKey: object cannot be nil (key: branch_identity)`

Relevant stack trace (from a symbolicated release build crash):

```
    0   CoreFoundation                  0x27a710d2 __exceptionPreprocess + 122
    1   libobjc.A.dylib                 0x35dcbc72 objc_exception_throw + 34
    2   CoreFoundation                  0x2798c380 -[__NSDictionaryM setObject:forKey:] + 856
    3   NativLabsTouchAppVieweriOS      0x00080a36 +[BNCFabricAnswers prepareBranchDataForAnswers:] (BNCFabricAnswers.m:46)
    4   NativLabsTouchAppVieweriOS      0x0008059e +[BNCFabricAnswers sendEventWithName:andAttributes:] (BNCFabricAnswers.m:16)
    5   NativLabsTouchAppVieweriOS      0x00075866 -[BranchOpenRequest processResponse:error:] (BranchOpenRequest.m:111)
    6   NativLabsTouchAppVieweriOS      0x00067406 __30-[Branch processNextQueueItem]_block_invoke (Branch.m:1293)
    7   libdispatch.dylib               0x3635bc7e _dispatch_call_block_and_release + 6
```

This only happens on our iOS 8 device (iOS 9 + 10, and Android are fine). It appears to occur sometime after `BranchIOS.Init` is called and our main View Controller is loaded and displayed.

I've noticed that the iOS SDK that is wrapped by these bindings are 0.12.3 based on the debug output. The native iOS library is up to version 0.12.11, but I didn't see anything addressing this directly, however there are a number of crash fixes.

Best I can tell from looking at the source code from [the iOS repo](https://github.com/BranchMetrics/ios-branch-deep-linking) is that because the App has never been launched before the PreferenceHelper doesn't have a value for 'identityId' yet. I also have a symbolicated crash report that gives file names and line numbers in the iOS project.

We've been debating on dropping iOS 8 support soon anyways, but my limited glance at the code doesn't seem to suggest that an iOS version would fix the issue. At best it appears to be dependent on the order the items `[Branch processNextQueueItem]` are inserted/processed.

**Comment by Taborke** [comment]
We are having the exact same issue. The second time the deep link is clicked and app is opened, it works as expected. All iOS 9+ devices and Android work. We are confused since we implemented this a few months back with the latest nuget library and now suddenly branch crashes the app for iOS 8 devices.

**Comment by aaustin** [comment]
Hey @Bobcat64 - Thanks so much for the complete crash report here. This seems to be a recent crash report for both of you? Do you have any info about when it started? We haven't changed anything with this library in quite some time...

**Comment by antonargunov** [comment]
Hey @Bobcat64 - We have uploaded new version of Xamarin SDK.
Read please documentation, because there are several changes in new SDK.
Can you try our new version of SDK?

**Comment by Bobcat64** [comment]
@aaustin @antonargunov Thanks for the responses and apologies for not getting back sooner.

We decided to drop iOS 8 support so this has become a non-issue for us. We also upgraded our iOS devices so I cannot easily test this on an iOS 8 environment.

I'll see about migrating to the new SDK, for now I'll close this issue as it is likely a fluke, will not be much of an issue since iOS 8 is disappearing, and reproducing and testing it will likely be more work than its worth.
